test/xbgpu/test_bsend.py

```diff
@@ -0,0 +1,228 @@
+################################################################################
+# Copyright (c) 2023, National Research Foundation (SARAO)
+#
+# Licensed under the BSD 3-Clause License (the "License"); you may not use
+# this file except in compliance with the License. You may obtain a copy
+# of the License at
+#
+#   https://opensource.org/licenses/BSD-3-Clause
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+"""Unit tests for the :mod:`katgpucbf.xbgpu.bsend` module."""
+
+from typing import Final, Sequence
+
+import numpy as np
+import pytest
+import spead2
+import spead2.recv.asyncio
+import spead2.send.asyncio
+from aiokatcp import Sensor, SensorSet
+from katsdpsigproc.abc import AbstractContext
+from katsdptelstate.endpoint import Endpoint
+
+from katgpucbf import COMPLEX
+from katgpucbf.spead import BF_RAW_ID, FREQUENCY_ID, TIMESTAMP_ID
+from katgpucbf.utils import TimeConverter
+from katgpucbf.xbgpu.bsend import BSend
+from katgpucbf.xbgpu.output import BOutput
+
+from . import test_parameters
+
+HEAPS_PER_FENG_PER_CHUNK: Final[int] = 5
```

Comment:
I don't think this is really the right name for this context. BSend has nothing to do with F-engines.

Reply:
In case this gets lost in the updates, I updated to `HEAPS_PER_CHUNK`.